File: ros2_ws/src/arduinobot_py_examples/arduinobot_py_examples/simple_subscriber.py
import rclpy

from rclpy.node import Node
from std_msgs.msg import String


class SimpleSubscriberNode(Node):
    def __init__(self):
        super().__init__("simple_subscriber")
        self.subs_ = self.create_subscription(String, "chatter", self.messageCallback, 10)

    def messageCallback(self, msg: String):
        self.get_logger().info(f"I heard: {msg.data}")


def main():
    rclpy.init()
    simpleSubscriber = SimpleSubscriberNode()
    rclpy.spin(simpleSubscriber)
    simpleSubscriber.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: ros2_ws/src/arduinobot_cpp_examples/src/simple_subscriber.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

using std::placeholders::_1;

class SimpleSubscriberNode : public rclcpp::Node
{
public:
    SimpleSubscriberNode() : Node("simple_subscriber")
    {
        sub_ = create_subscription<std_msgs::msg::String>("chatter", 10, std::bind(&SimpleSubscriberNode::msgCallback, this, _1));
    }

private:
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_;

    void msgCallback(const std_msgs::msg::String msg) const{
        RCLCPP_INFO_STREAM(get_logger(), "I heard: " << msg.data.c_str());
    }
};


int main(int argc, char * argv[]){
    rclcpp::init(argc, argv);
    auto node =  std::make_shared<SimpleSubscriberNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}